### PR TITLE
fix: covalent tx parse

### PR DESCRIPTION
**Pull request by sidmorizon**

### CODE REVIEW

**Review by revan-zhang — APPROVED** [review]

**Review by sunnylqm — APPROVED** [review]

**Review by taimanhui — APPROVED** [review]

File: packages/engine/src/vaults/types.ts [hunk 1]
```diff
@@ -270,25 +271,33 @@ export type IDecodedTxActionTokenApprove = IDecodedTxActionBase & {
...
+  value: string;
+  data?: string;
+};
 export type IDecodedTxActionInternalSwap = IDecodedTxActionBase & ISwapInfo;
 // other Unknown Action
 export type IDecodedTxActionUnknown = IDecodedTxActionBase;
 export type IDecodedTxAction = {
   type: IDecodedTxActionType;
   direction?: IDecodedTxDirection; // TODO move direction to UI side generate
+  hidden?: boolean;
```
Comment (sidmorizon) [inline packages/engine/src/vaults/types.ts:292]: isHidden

File: packages/engine/src/vaults/types.ts [hunk 2]
```diff
@@ -270,25 +271,33 @@ export type IDecodedTxActionTokenApprove = IDecodedTxActionBase & {
...
   nativeTransfer?: IDecodedTxActionNativeTransfer;
   tokenTransfer?: IDecodedTxActionTokenTransfer;
   tokenApprove?: IDecodedTxActionTokenApprove;
   internalSwap?: IDecodedTxActionInternalSwap;
   functionCall?: IDecodedTxActionFunctionCall;
   // other Unknown Action
   unknownAction?: IDecodedTxActionUnknown;
+  evmInfo?: IDecodedTxActionEvmInfo;
```
Comment (sidmorizon) [inline packages/engine/src/vaults/types.ts:300]: actionInfoEvm

File: packages/engine/src/vaults/types.ts [hunk 3]
```diff
@@ -243,8 +243,9 @@ export type IDecodedTxActionBase = {
 
 export type IDecodedTxActionFunctionCall = IDecodedTxActionBase & {
   target: string; // contractAddress
-  functionName: string;
-  functionHash: string; // functionSignature
+  functionName: string; // approve
+  functionHash?: string; // 0x095ea7b3
+  functionSignature?: string; //
```
Comment (sidmorizon) [inline packages/engine/src/vaults/types.ts:248]: 注释

File: packages/engine/src/vaults/types.ts [hunk 4]
```diff
@@ -228,7 +228,7 @@ export enum IDecodedTxActionType {
   // SWAP = 'SWAP',
 
   // Contract Interaction
-  // FUNCTION_CALL = 'FUNCTION_CALL',
+  FUNCTION_CALL = 'FUNCTION_CALL',
 
   // other
   TRANSACTION = 'TRANSACTION',
```
Comment (sidmorizon) [inline packages/engine/src/vaults/types.ts:234]: UNKNOWN

File: packages/engine/src/vaults/types.ts (hunk 3 above)
Comment (sidmorizon) [reply packages/engine/src/vaults/types.ts:248]: "Withdrawal(indexed address src, uint256 wad)"

**Pull request merged**